### PR TITLE
feat: adds foundry-zksync docs

### PR DESCRIPTION
**Pull request by dutterbutter**
# What :computer: 
* Adds foundry documentation for alpha release: https://github.com/matter-labs/foundry-zksync/releases/tag/v0.0.1-alpha.1

# Why :hand:
* Provides overview, setup, usage guides

# Evidence :camera:
Include screenshots, screen recordings, or `console` output here demonstrating that your changes work as intended

Screenshot 2024-01-26 at 2 51 44 PM

# Notes :memo:
* Any notes/thoughts that the reviewers should know prior to reviewing the code?

### CODE REVIEW

File: docs/build/test-and-debug/foundry.md [hunk 1]
```diff
@@ -5,10 +5,240 @@ head:
...
+
+**Usage:**
+
+```sh
+zkforge zk-build [OPTIONS]
+```
+
+**Key Compiler Options:**
+
+- `--use-zksolc <ZK_SOLC_VERSION>`: Specify the solc version or a local solc path.
```
Comment (uF4No) [inline docs/build/test-and-debug/foundry.md:90]: is this the solc or zksolc version?

Comment (dutterbutter) [reply docs/build/test-and-debug/foundry.md:90]: zksolc good 👁️

File: docs/build/tooling/foundry/overview.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  - - meta
+    - name: "twitter:title"
+      content: Foundry | zkSync Docs
+---
+
+# Foundry with zkSync
+
+## Overview
+
+`foundry-zksync` is a specialized fork of [Foundry](https://github.com/foundry-rs/foundry), tailored for zkSync. It extends Foundry's capabilities for Ethereum app development to support zkSync, allowing for the compilation, deployment, testing, and interaction with smart contracts on zkSync. `foundry-zksync` introduces `zkforge` and `zkcast` extensions of Foundry's existing `forge` and `cast` tools but tailored for zkSync usage.
```
Comment (MexicanAce) [inline docs/build/tooling/foundry/overview.md:12]: Perhaps this is something more for zkfoundry README, but we should document which upstream commit/version we forked

**Review by MexicanAce — APPROVED** [review]
LGTM

**Review by MexicanAce — APPROVED** [review]

File: docs/build/tooling/foundry/overview.md (hunk 2 above)
Comment (dutterbutter) [reply docs/build/tooling/foundry/overview.md:12]: Let me make this change.

**Pull request merged**